include/psa/crypto_values.h

```diff
@@ -1779,6 +1779,17 @@
 #define PSA_ALG_PBKDF2_HMAC_GET_HASH(hkdf_alg)                         \
     (PSA_ALG_CATEGORY_HASH | ((hkdf_alg) & PSA_ALG_HASH_MASK))
 
+/** The PBKDF2-AES-CMAC-PRF-128 password hashing / key stretching algorithm.
+ *
+ * PBKDF2 is defined by PKCS#5, republished as RFC 8018 (section 5.2).
+ * This macro specifies the PBKDF2 algorithm constructed using the
+ * AES-CMAC-PRF-128 PRF specified by RFC 4615.
```

Comment:
I'm sorry, perhaps I'm reading RFC 6803 too quickly but I don't think it's about using using a Camellia-based construct as the PRF for PBKDF2, but rather about using PBKDF2-HMAC-SHA1 to derive Camellia keys. Actually, they even need to tweak the salt to achieve domain separation from AES keys, which they wouldn't need to do if they were changing the PBKDF2 PRF.

Before adding this identifier, I had similar reservations so I spent some time looking for uses of other PRFs in PBKDF2 in publicly-available standards, open source software and other publicly documented APIs, and couldn't find any other than the ones in this PR, which is the reason to target specifically AES-128.

(Of course that doesn't prove there aren't any, I could have missed them and some might even appear in the future - I would think nowadays people would turn to newer schemes, but the Thread spec would tend to prove me wrong.) But overall, I think the probabily that we end up introducing other algorithm identifiers in the future seems low, and even then as you note, we'd probably end up adding a couple specific names rather than a generic macro anyway.

So I'm inclined to  keep this identifier here. But if after considering my points you still feel it's best to defer, I'm OK with that too.

Reply:
Ah, yes, my bad: RFC 6803 uses a Camellia-CMAC-based non-stretching KDF to derive key material, but uses PBKDF2-HMAC-SHA1 to stretch.

So while I'm still a little uncomfortable with encoding one specific block cipher in an algorithm (because block ciphers are somewhat region-specific), I'm ok with the pragmatic choice of including this one because it's the only one that has a known deployed protocool using it.